### ISSUE TITLE
Cache is never populated in ComponentDesigner.ShadowPropertyCollection.cs

### ISSUE THREAD

**Issue by Jericho** [dotnet/winforms#8231]
### .NET version

8.0.0-alpha.1.22559.2

However, the situation I am reporting is not tied to a specific .NET version.

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

no

### Issue description

The ShadowPropertyCollection class contains a private field called `_descriptors` which, if I understand the original intent correctly, is supposed to be used as a cache of `PropertyDescriptor`. The flow is supposed to be like this:
- the first-time a given property descriptor is requested, the value is calculated in the `private PropertyDescriptor GetShadowedPropertyDescriptor(string propertyName)` private method
- this value is added to the cache
- if the same property descriptor is subsequently requested the cached value is returned

However, what I observe when reading the code of this private method is that the calculated value is NOT added to the cache and therefore it is re-calculated each time the same descriptor is requested.

@elachlan, @dreddy-work and myself had a [discussion](https://github.com/dotnet/winforms/pull/8222#pullrequestreview-1185033949) about this and we concluded that it made more sense to raise a separate issue about this rather than simply fix it as part of #8222.

If the conclusion is that we want to fix the flawed logic, I will be happy to volunteer a PR.

### Steps to reproduce

To be clear, the current logic works without problem and returns the expected result. It's just that we pay a performance penalty if the same descriptor is requested multiple times.

**Comment by elachlan** [comment]
Relevant code where `_descriptors` is initialized and checked, but never added to.
https://github.com/dotnet/winforms/blob/bc73fc2790fdce8ac7399eddf76824161bdd35cc/src/System.Windows.Forms.Design/src/System/ComponentModel/Design/ComponentDesigner.ShadowPropertyCollection.cs#L59-L67

Proposed fix by @Jericho:
```csharp
private PropertyDescriptor GetShadowedPropertyDescriptor(string propertyName)
{
    _descriptors ??= new();

    if (!_descriptors.TryGetValue(propertyName, out PropertyDescriptor property))
    {
        property = TypeDescriptor.GetProperties(_designer.Component.GetType())[propertyName];
        _descriptors.Add(propertyName, property);
    }

    return property;
}
```

**Comment by elachlan** [comment]
@dreddy-work how does the team feel about this change?

**Comment by dreddy-work** [comment]
> @dreddy-work how does the team feel about this change?

This is scheduled for a quick check within the team. You won't hear until Next week. Will update this thread after.

**Comment by dreddy-work** [comment]
We haven't got chance to go over this yet. It may have to wait for some more time.

**Comment by elachlan** [comment]
@dreddy-work should you add the "waiting on team" label, just in case this gets lost to time?

**Comment by dreddy-work** [comment]
> @dreddy-work should you add the "waiting on team" label, just in case this gets lost to time?

This one is waiting on me to collect more info around scenario here. We tried to go over it in last meeting but i needed more info around scenarios here. I will go over it early in the new year as i am going on vacation from this thursday.

**Comment by elachlan** [comment]
No problem! Enjoy your holidays :)

**Comment by merriemcgaw** [comment]
We'd like to do a bit of investigation on the VS side of things so we can make sure that we don't run into something unexpected.

**Comment by elachlan** [comment]
@merriemcgaw can the team slate this for .NET9?

**Comment by merriemcgaw** [comment]
@elachlan absolutely!

**Comment by lonitra** [comment]
This needs to be re-assigned.
